### PR TITLE
Remove src_postgres

### LINKED ISSUE #2688: src_postgres needs to be deprecated everywhere
### Bug Description

`src_postgres` usage is deprecated but looks like it was not grepped and replaced everywhere. It still lives in the codebase in some places, e.g. [1](https://github.com/PecanProject/pecan/blob/develop/modules/data.land/R/get.veg.module.R#L32), [2](https://github.com/PecanProject/pecan/blob/d962b2053e42bc0a2a7bcd4339bd4a15b76e91e3/modules/data.land/R/soil_process.R#L29), ...

This is probably fine for majority of pecan users for now. But, for example, I use another port for the DB connection and these codes will not like that. 

It is probably good to be consistent in how we open DB connection throughout the codebase anyway.

This is an issue to remind us to replace the remaining ones as well.

### PR DESCRIPTION
**Pull request by infotroph**
This is a continuation of PR #2785, which we can no longer update because of an accidentally deleted source branch that turned out to be more hassle to fix than seemed worth it. All the commits from #2785 are contained here and all comments from that discussion thread are now resolved.

## Description

* Removed all remaining calls to `dplyr::src_postgres`, mostly by replacing with `PEcAn.DB::db.open()` but sometimes with `DBI::dbConnect()` in places that are run without the PEcAn packages available. Fixes #2688 
* Converted all `bety <- db.open(...); con <- bety$con` to `con <- db.open(...)`
* Converted all `bety$con` to just `bety` (or `con`, in cases where `bety` was renamed in previous step) 
* Converted most existing `DBI::dbConnect(...)` calls to `PEcAn.DB::db.open(...)`, except in contexts where PEcAn.DB may not be available (e.g. pecanapi demo) or to access non-PEcAn databases (e.g. some examples in DB package)
* Consistently pass all of `settings$database$bety` into `db.open()` rather than pick out individual components. Attn @istfer: I believe this should allow port specification to work consistently as requested in https://github.com/PecanProject/pecan/issues/2688#issuecomment-898333223

## Motivation and Context

`dplyr::src_postgres()` is now deprecated and the advice is to instead directly pass a DBI connection into `dplyr::tbl()`. We've previously replaced a few of these piecemeal, but not systematically (as evidenced by all the places the code was handling both bare `con` and `bety$con`).

## Review Time Estimate

- [ ] Immediately
- [ ] Within one week
- [ ] When possible

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue) 
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My change requires a change to the documentation.
- [ ] My name is in the list of .zenodo.json
- [x] I have updated the CHANGELOG.md.
- [x] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by moki1202 — APPROVED** [review]

**Comment by dlebauer** [comment]
Did you update the changelog?--
David

-- 

David LeBauer
Director of Data Sciences
CALS Comm and Technologies
THE UNIVERSITY OF ARIZONA

BSRL, 207
1230 N Cherry Ave | Tucson, AZ 85721
Office: 520-621-4381 | Cell: 760-468-8621
***@***.***

datascience.cals.arizona.edu
orcid | twitter | github | linkedin

**Comment by infotroph** [comment]
> Did you update the changelog?
@dlebauer Yes! Thanks for checking. https://github.com/PecanProject/pecan/pull/2881/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4edR13

File: modules/data.land/R/ens.veg.module.R [hunk 1]
```diff
@@ -10,58 +10,53 @@
...
   # Write model specific IC files
-  bety <- dplyr::src_postgres(dbname   = dbparms$bety$dbname, 
-                              host     = dbparms$bety$host, 
-                              user     = dbparms$bety$user, 
-                              password = dbparms$bety$password)
-  
-  con <- bety$con
-  on.exit(db.close(con), add = TRUE)
-  
+  con <- PEcAn.DB::db.open(dbparms)
```
Comment (istfer) [inline modules/data.land/R/ens.veg.module.R:28]: should these be:
```suggestion
  con <- PEcAn.DB::db.open(dbparms$bety)
```

File: modules/data.land/R/get.veg.module.R [hunk 2]
```diff
@@ -9,76 +9,72 @@
...
   site_name <- new_site$name
   ## Prepare to call convert.inputs
   pkg  <- "PEcAn.data.land"
-  bety <- dplyr::src_postgres(dbname   = dbparms$bety$dbname, 
-                              host     = dbparms$bety$host, 
-                              user     = dbparms$bety$user, 
-                              password = dbparms$bety$password)
-  con  <- bety$con
-  
+  con <- PEcAn.DB::db.open(dbparms)
```
Comment (istfer) [inline modules/data.land/R/get.veg.module.R:32]:
```suggestion
  con <- PEcAn.DB::db.open(dbparms$bety)
```

File: modules/data.remote/inst/download_SMAP_gee2pecan.R [hunk 3]
```diff
@@ -35,10 +35,9 @@ download_SMAP_gee2pecan <- function(start, end,
   
   #################### Connect to BETY #################### 
   
-  bety <- list(user='bety', password='bety', host='localhost',
-               dbname='bety', driver='PostgreSQL',write=TRUE)
-  con <- PEcAn.DB::db.open(bety)
-  bety$con <- con
+  con <- PEcAn.DB::db.open(
```
Comment (istfer) [inline modules/data.remote/inst/download_SMAP_gee2pecan.R:38]: I guess there isn't much to be done with these hardcoded ones

File: modules/data.land/R/ens.veg.module.R (hunk 1 above)
Comment (infotroph) [reply modules/data.land/R/ens.veg.module.R]: Good catch, thanks!

**Review by istfer — APPROVED** [review]

**Comment by infotroph** [comment]
@robkooper Can you squash-merge this, please? The commits contain a lot of back and forth that doesn't need to be preserved.

**Comment by robkooper** [comment]
done

**Pull request merged**